diff --git a/node_modules/sendpulse-api/api/sendpulse.js b/node_modules/sendpulse-api/api/sendpulse.js
index 4d34504..7b2c538 100644
--- a/node_modules/sendpulse-api/api/sendpulse.js
+++ b/node_modules/sendpulse-api/api/sendpulse.js
@@ -449,13 +449,20 @@ function getEmailsFromBook(callback, id) {
  * @param callback
  * @param id
  * @param emails
+ * @params sender
+ * @params template_id
  */
-function addEmails(callback, id, emails) {
+function addEmails(callback, id, emails, sender, template_id) {
     if ((id === undefined) || (emails === undefined) || (!emails.length)) {
         return callback(returnError('Empty email or book id'));
+    } else {
+        if ((sender === undefined) || (template_id === undefined)) {
+            var data = {emails: serialize(emails)};
+        } else {
+            var data = {emails: serialize(emails), confirmation: "force", sender_email: sender, template_id: template_id, message_lang: "en"};
+        }
+        sendRequest('addressbooks/' + id + '/emails', 'POST', data, true, callback);
     }
-    var data = {emails: serialize(emails)};
-    sendRequest('addressbooks/' + id + '/emails', 'POST', data, true, callback);
 }
 
 /**
diff --git a/node_modules/sendpulse-api/example.js b/node_modules/sendpulse-api/example.js
index ea9ae42..bb6463e 100644
--- a/node_modules/sendpulse-api/example.js
+++ b/node_modules/sendpulse-api/example.js
@@ -38,7 +38,7 @@ sendpulse.init(API_USER_ID, API_SECRET, TOKEN_STORAGE, function(token) {
     sendpulse.removeAddressBook(answerGetter, 123456);
     sendpulse.getBookInfo(answerGetter,123456);
     sendpulse.getEmailsFromBook(answerGetter,123456);
-    sendpulse.addEmails(answerGetter, 123456, [{email:'some@domain.com',variables:{}}]);
+    sendpulse.addEmails(answerGetter, 123456, [{email:'some@domain.com',variables:{}}], 'sender@domain.com', 'template_id');
     sendpulse.removeEmails(answerGetter, 123456, ['some@domain.com']);
     sendpulse.getEmailInfo(answerGetter,123456,'some@domain.com');
     sendpulse.campaignCost(answerGetter,123456);
diff --git a/node_modules/sendpulse-api/index.d.ts b/node_modules/sendpulse-api/index.d.ts
index a0d9380..e1acef2 100644
--- a/node_modules/sendpulse-api/index.d.ts
+++ b/node_modules/sendpulse-api/index.d.ts
@@ -2,7 +2,7 @@
 
 export function activateSender(callback: any, senderEmail: any, code: any): any;
 
-export function addEmails(callback: any, id: any, emails: any): any;
+export function addEmails(callback: any, id: any, emails: any, sender: any, template_id: any): any;
 
 export function addSender(callback: any, senderName: any, senderEmail: any): any;
 
